### ISSUE TITLE
options.bootClasspath build cache key differs on Windows and Linux

### ISSUE THREAD

**Issue by diacone** [gradle/gradle#2619]
### Expected Behavior
The build chache is hit even if source/target build platform differs

### Current Behavior
The build cache is is not hit.

### Context
When you set CompileOptions.bootClasspath you must use platform specific path separator (; vs :) and the cache key differs betwen platforms (windows and linux).

### Steps to Reproduce (for bugs)
Set CompileOptions.bootClasspath to something with multiple paths and run the build on windows and on linux:
```
def boot = project.fileTree("/opt/jdk/jre/lib") { include '*.jar' }
def bootcp = (boot.files as SortedSet).join(File.pathSeparator)

tasks.withType(AbstractCompile) {
  options.bootClasspath = bootcp
}

```

### Your Environment
* Oracle Linux 6 , Fedora 25, Windows 10
* Gradle 4.0.2

**Comment by bmuschko** [comment]
@lptr I see you already referenced this issue in a potential fix. Could you please provide an outlook on the Gradle version that would fix the issue?

**Comment by lptr** [comment]
The plan is to address this for 4.2.

**Comment by lptr** [comment]
Unfortunately, this won't make it into 4.2.

**Comment by mark-vieira** [comment]
This problem is actually more than just os-specific path separators. The only way to configure this option is to provide an absolute path string which means you'll get cache misses even on the same OS if the JARs you're pointing to live in different locations.

**Comment by lptr** [comment]
This is now fixed as #3002 and will ship in Gradle 4.3.
